### ISSUE TITLE
Other collections on stack

### ISSUE THREAD

**Issue by rogday** [slightlyoutofphase/staticvec#33]
Are you planning on adding other collections? Like HashMap/HashSet, VecDeque, BinaryHeap? I found that "heapless" crate providing BinaryHeap on stack is slow for some reason. I tried copying BinaryHeap from std and changing Vec to StaticVec(and removing a lot of stuff that I didn't care), worked like a charm!

**Comment by slightlyoutofphase** [comment]
That's an eventual goal, yeah.  I've been focusing on getting `StaticString` up to where I want it for the past couple weeks or so, but I'll likely start looking at some of that stuff next.

**Comment by slightlyoutofphase** [comment]
I'm now intending to start working on some of this kind of stuff soon, so I wanted to ask if whatever you have as far as `BinaryHeap` is something you'd be interesting in contributing? Even if it's just a "skeleton" implementation, I'd still be happy to merge anything that might save me a bit of time by giving me something to work off of!

If you're interested, feel free to just go ahead and open a PR.

**Comment by rogday** [comment]
I'm not sure I know Rust enough for PR into the crate(playing around with it for a few weeks). Although I  think you can copy some of my copy-pasting: https://pastebin.com/rNqyUnag
That's basically https://github.com/rust-lang/rust/blob/master/src/liballoc/collections/binary_heap.rs but with StaticVec istead of Vec, check with diff to know for sure that I didn't break anything
TLDR: docs aren't fixed and I removed all extend/append methods because of lack of reserve method

**Comment by slightlyoutofphase** [comment]
I'll take a look to see if there's anything I can make use of. Thanks!

**Comment by slightlyoutofphase** [comment]
I've just committed an initial version of `StaticHeap`. Your pastebin definitely helped, so thanks a lot!

Will close this issue now, I guess. (If you're wondering , I am still planning to continue to add more collection types after `StaticHeap`, although I'm not sure which ones quite yet.)
